### ISSUE TITLE
Scylla crashed with "/builddir/build/BUILD/scylla-3.0.rc3/seastar/core/iostream.hh:299: seastar::output_stream<CharType>::~output_stream() [with CharType = char]: Assertion `!_in_batch' failed."

### ISSUE THREAD

**Issue by yarongilor** [scylladb/scylla#4025]
*Installation details*
Scylla version (or git commit hash): 3.0.rc3-0.20181224.5558fa8c4
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): AWS AMI

Scylla crashed with coredump during longevity test with the following error and backtrace:
```
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: scylla: /builddir/build/BUILD/scylla-3.0.rc3/seastar/core/iostream.hh:299: seastar::output_stream<CharType>::~output_stream() [with CharType = char]: Assertion `!_in_batch' failed.
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: Aborting on shard 2.
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: Backtrace:
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000006a7732
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000005bee5c
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000005bf105
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000005bf1b3
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libpthread.so.0+0x000000000000f5cf
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libc.so.6+0x0000000000036206
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libc.so.6+0x00000000000378f7
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libc.so.6+0x000000000002f025
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libc.so.6+0x000000000002f0d1
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000008dce4d
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x0000000003293019
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x0000000003207e46
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000038ac735
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000005a020e
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000005a03d1
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x000000000066ea67
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x00000000006731da
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: 0x000000000075626d
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libpthread.so.0+0x0000000000007dd4
Dec 26 11:03:13 ip-172-16-99-232.ec2.internal scylla[51103]: /lib64/libc.so.6+0x00000000000fdeac
Dec 26 11:04:51 ip-172-16-99-232.ec2.internal scylla[62013]: Scylla version 3.0.rc3-0.20181224.5558fa8c4 starting ...

```

Decoded Backtrace:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/src/debug/scylla-3.0.rc3/seastar/util/backtrace.hh:56
seastar::backtrace_buffer::append_backtrace() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:410
 (inlined by) print_with_backtrace at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:431
print_with_backtrace at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:438
sigabrt_action at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4024
 (inlined by) operator() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4006
 (inlined by) _FUN at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4002
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
seastar::output_stream<char>::~output_stream() at /usr/src/debug/scylla-3.0.rc3/seastar/core/iostream.hh:299
 (inlined by) seastar::rpc::connection::~connection() at /usr/src/debug/scylla-3.0.rc3/seastar/rpc/rpc.hh:245
seastar::rpc::client::~client() at /usr/src/debug/scylla-3.0.rc3/seastar/rpc/rpc.hh:313
 (inlined by) seastar::shared_ptr_count_for<seastar::rpc::client>::~shared_ptr_count_for() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:463
 (inlined by) seastar::shared_ptr_count_for<seastar::rpc::client>::~shared_ptr_count_for() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:463
seastar::shared_ptr<seastar::rpc::connection>::~shared_ptr() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:535
 (inlined by) seastar::foreign_ptr<seastar::shared_ptr<seastar::rpc::connection> >::~foreign_ptr() at /usr/src/debug/scylla-3.0.rc3/seastar/core/sharded.hh:615
 (inlined by) seastar::shared_ptr_no_esft<seastar::foreign_ptr<seastar::shared_ptr<seastar::rpc::connection> > >::~shared_ptr_no_esft() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:157
 (inlined by) seastar::internal::lw_shared_ptr_accessors_no_esft<seastar::foreign_ptr<seastar::shared_ptr<seastar::rpc::connection> > >::dispose(seastar::lw_shared_ptr_counter_base*) at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:213
 (inlined by) seastar::lw_shared_ptr<seastar::foreign_ptr<seastar::shared_ptr<seastar::rpc::connection> > >::~lw_shared_ptr() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:300
 (inlined by) seastar::rpc::sink<frozen_mutation_fragment>::impl::~impl() at /usr/src/debug/scylla-3.0.rc3/seastar/rpc/rpc_types.hh:256
 (inlined by) seastar::rpc::sink_impl<netw::serializer, frozen_mutation_fragment>::~sink_impl() at /usr/src/debug/scylla-3.0.rc3/seastar/rpc/rpc.hh:298
 (inlined by) seastar::shared_ptr_count_for<seastar::rpc::sink_impl<netw::serializer, frozen_mutation_fragment> >::~shared_ptr_count_for() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:463
 (inlined by) seastar::shared_ptr_count_for<seastar::rpc::sink_impl<netw::serializer, frozen_mutation_fragment> >::~shared_ptr_count_for() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:463
seastar::shared_ptr<seastar::rpc::sink<frozen_mutation_fragment>::impl>::~shared_ptr() at /usr/src/debug/scylla-3.0.rc3/seastar/core/shared_ptr.hh:535
 (inlined by) seastar::rpc::sink<frozen_mutation_fragment>::~sink() at /usr/src/debug/scylla-3.0.rc3/seastar/rpc/rpc_types.hh:247
 (inlined by) seastar::internal::do_with_state<seastar::rpc::sink<frozen_mutation_fragment>, seastar::future<> >::~do_with_state() at /usr/src/debug/scylla-3.0.rc3/seastar/core/do_with.hh:48
 (inlined by) seastar::internal::do_with_state<seastar::rpc::sink<frozen_mutation_fragment>, seastar::future<> >::~do_with_state() at /usr/src/debug/scylla-3.0.rc3/seastar/core/do_with.hh:48
 (inlined by) seastar::internal::do_with_state<seastar::rpc::sink<frozen_mutation_fragment>, seastar::future<> >::run_and_dispose() at /usr/src/debug/scylla-3.0.rc3/seastar/core/do_with.hh:55
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:2695
seastar::reactor::run_some_tasks() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:3118
seastar::reactor::run_some_tasks() at /usr/src/debug/scylla-3.0.rc3/seastar/util/log.hh:313
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:3265
operator() at /usr/src/debug/scylla-3.0.rc3/seastar/core/reactor.cc:4334
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-3.0.rc3/seastar/core/posix.cc:52

```

**Comment by gleb-cloudius** [comment]
On Wed, Dec 26, 2018 at 07:30:08AM -0800, yarongilor wrote:
> @gleb-cloudius, core-dump is temporarily available in 172.16.99.232 (public IP available upon request).
> 
Do you have logs from 172.16.99.10 for DEC 25?

--
			Gleb.

**Comment by gleb-cloudius** [comment]
On Thu, Dec 27, 2018 at 10:37:07AM +0200, Gleb Natapov wrote:
> On Wed, Dec 26, 2018 at 07:30:08AM -0800, yarongilor wrote:
> > @gleb-cloudius, core-dump is temporarily available in 172.16.99.232 (public IP available upon request).
> > 
> Do you have logs from 172.16.99.10 for DEC 25?
> 
No need, the core is from dec 26.

--
			Gleb.
